rest_framework/permissions.py

```diff
@@ -228,8 +228,13 @@ def has_permission(self, request, view):
 
         queryset = self._queryset(view)
         perms = self.get_required_permissions(request.method, queryset.model)
+        change_perm = self.get_required_permissions('PUT', queryset.model)
+
+        user = request.user
+        if request.method == 'GET':
+            return user.has_perms(perms) or user.has_perms(change_perm)
```

Comment:
I don't see the motivation for this additional set of checks against `view_*` or `change_*` permissions.

Just seems inconsistent to me.

Reply:
this maintains backward compatibility Tom. If the user would miss the change permission but the DB was not updated to grant the user view permissions (because the DB was not updated), the app will break because users have no view permissions assigned, only change.